### PR TITLE
Adding lift from graph to Vietoris-Rips complex.

### PR DESCRIPTION
**Pull request by rballeba**
This commit creates a method to convert a graph into a Vietoris-Rips clique complex. It mimics the behaviour of the previous method graph_to_clique_complex but only adds cliques whose nodes have pairwise distances less than a radius parameter of the function. Unit tests added.

### CODE REVIEW

File: toponetx/transform/graph_to_simplicial_complex.py [hunk 1]
```diff
@@ -93,3 +95,45 @@ def graph_2_clique_complex(
...
+    def is_in_vr_complex(clique):
+        edges = combinations(clique, 2)
+        edge_weights_lower_than_r = all(G[u][v]["weight"] <= r for u, v in edges)
+        return edge_weights_lower_than_r
+
+    all_cliques = nx.enumerate_all_cliques(G)
+    possible_cliques = (
+        all_cliques
+        if max_dim is None
+        else filter(lambda face: len(face) <= max_dim, all_cliques)
```
Comment (ffl096) [inline toponetx/transform/graph_to_simplicial_complex.py:136]: Please don't use `filter` here, it is inefficient for low `max_dim`. See `graph_to_clique_complex` in this file and #221.

**Review by ffl096 — CHANGES_REQUESTED** [review]

File: toponetx/transform/graph_to_simplicial_complex.py [hunk 2]
```diff
@@ -29,6 +30,7 @@ def graph_to_neighbor_complex(G: nx.Graph) -> SimplicialComplex:
     -----
     This type of simplicial complexes can have very large dimension ( dimension = max_i(len (G.neighbors(i))) )
     and it is a function of the distribution of the valency of the graph.
+
```
[the remark this answers is not in the document]
Comment (rballeba) [reply toponetx/transform/graph_to_simplicial_complex.py]: Done

File: toponetx/transform/graph_to_simplicial_complex.py (hunk 1 above)
Comment (rballeba) [reply toponetx/transform/graph_to_simplicial_complex.py]: Done

File: toponetx/transform/graph_to_simplicial_complex.py [hunk 3]
```diff
@@ -93,3 +95,47 @@ def graph_2_clique_complex(
         stacklevel=2,
     )
     return graph_to_clique_complex(G, max_dim)
+
+
+def weighted_graph_to_vietoris_rips_complex(
+    G: nx.Graph, r: float, max_dim: Optional[int] = None
```
Comment (ffl096) [inline toponetx/transform/graph_to_simplicial_complex.py:101]:
```suggestion
    G: nx.Graph, r: float, max_dim: int | None = None
```

**Review by ffl096 — CHANGES_REQUESTED** [review]
Otherwise looks good to me!

**Comment by mhajij** [comment]
@ffl096 do you request any new changes ?

**Comment by ffl096** [comment]
Perfect, thank you!

**Pull request merged**